packages/babylon/src/tokenizer/index.js

```diff
@@ -827,13 +827,19 @@ export default class Tokenizer extends LocationParser {
     }
     const content = this.input.slice(start, this.state.pos);
     ++this.state.pos;
-    // Need to use `readWord1` because '\uXXXX' sequences are allowed
-    // here (don't ask).
-    const mods = this.readWord1();
-    if (mods) {
-      const validFlags = /^[gmsiyu]*$/;
-      if (!validFlags.test(mods)) {
-        this.raise(start, "Invalid regular expression flag");
+
+    const validFlags = /^[gmsiyu]*$/;
+    let mods = "";
+    while (this.state.pos < this.input.length) {
+      const char = this.input[this.state.pos];
+      const charCode = this.fullCharCodeAtPos();
+      if (validFlags.test(char)) {
+        ++this.state.pos;
+        mods += char;
+      } else if (!validFlags.test(mods) || charCode === charCodes.backslash) {
```

Comment:
`validFlags.test(mods)`  is always `true`, because we add a character to `mods` only if it is valid.
I think that this condition should be
```js
else if (isIdentifierChar(char)) {
```
so that it throws `Invalid regular expession flag` also for things like `/foo/w` (Instead of `Unexpected token, expected ";"`).

Reply:
RIght, so that means `core/uncategorized/108` and `core/uncategorized/366' should also throw `Invalid regular expression flags` yeah ?

Reply:
Yes :+1: